mcs/class/System.Net.Http/System.Net.Http/HttpClient.wasm.cs

```diff
@@ -9,41 +9,14 @@ public partial class HttpClient
 
 #pragma warning disable 649
         private static Func<HttpMessageHandler> GetHttpMessageHandler;
-        // Used to skip the type lookup on later calls.
-        private static MethodInfo messageHandlerMethod;
 #pragma warning restore 649
 
         internal static HttpMessageHandler CreateDefaultHandler()
         {
 
             if (GetHttpMessageHandler == null)
             {
-                if (messageHandlerMethod == null) {
-                    try {
-                        Type type = Assembly.Load("WebAssembly.Net.Http")?.GetType("WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler");
-                        if (type == null)
-                            throw new PlatformNotSupportedException ("Invalid WebAssembly Module? Cannot find WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler");
-
-                        messageHandlerMethod = type.GetMethod("GetHttpMessageHandler", BindingFlags.Static | BindingFlags.NonPublic);
-                    }
-                    catch { 
-                        throw new PlatformNotSupportedException ("Invalid WebAssembly Module? Cannot find WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler");
-                    }
-                }
-
-                if (messageHandlerMethod == null)
-                    throw new PlatformNotSupportedException ("Your WebAssembly version does not support obtaining of the custom HttpClientHandler");
-
-                object ret = messageHandlerMethod.Invoke(null, null);
-                if (ret == null)
-                    throw new PlatformNotSupportedException ("WebAssembly returned no custom HttpClientHandler");
-
-                var handler = ret as HttpMessageHandler;
-                if (handler == null)
-                    throw new PlatformNotSupportedException ($"{ret?.GetType()} is not a valid HttpMessageHandler");
-
-                return handler;
-
+                return WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler.GetHttpMessageHandler();
             }
             else
             {
```

Comment:
Where is this used?

Reply:
Introduced a `WebAssembly.RuntimeOptions` modeled after XI/XA.